<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabbar_function">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="Base">
       <attribute name="title">
        <string>Base</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout" stretch="1,6,2">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="topMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,0">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <property name="leftMargin">
           <number>6</number>
          </property>
          <item>
           <widget class="QComboBox" name="cmb_mode">
            <property name="toolTip">
             <string>Mode</string>
            </property>
            <item>
             <property name="text">
              <string>Serial</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>TCP Server</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>TCP Client</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>UDP</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTabWidget" name="tabbar_mode">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Ignored">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="tabPosition">
             <enum>QTabWidget::West</enum>
            </property>
            <property name="tabShape">
             <enum>QTabWidget::Rounded</enum>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <property name="elideMode">
             <enum>Qt::ElideNone</enum>
            </property>
            <property name="usesScrollButtons">
             <bool>false</bool>
            </property>
            <property name="documentMode">
             <bool>true</bool>
            </property>
            <property name="tabsClosable">
             <bool>false</bool>
            </property>
            <property name="tabBarAutoHide">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="tab_mode_serial">
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <attribute name="title">
              <string>UDP</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_2" columnstretch="6,2,1">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item row="0" column="0">
               <widget class="QComboBox" name="cmb_serial_port">
                <property name="toolTip">
                 <string>Port</string>
                </property>
                <property name="styleSheet">
                 <string notr="true"/>
                </property>
                <property name="sizeAdjustPolicy">
                 <enum>QComboBox::AdjustToContents</enum>
                </property>
               </widget>
              </item>
              <item row="0" column="1" colspan="2">
               <widget class="QComboBox" name="cmb_serial_baud">
                <property name="toolTip">
                 <string>BaudRate</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_mode_tcp_server">
             <attribute name="title">
              <string>tcp_server</string>
             </attribute>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QComboBox" name="cmb_server_ip"/>
              </item>
              <item>
               <widget class="QLineEdit" name="input_tcp_server_port">
                <property name="toolTip">
                 <string>Port</string>
                </property>
                <property name="text">
                 <string>50000</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_mode_tcp_client">
             <attribute name="title">
              <string>tcp_client</string>
             </attribute>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QComboBox" name="cmb_tcp_client_ip"/>
              </item>
              <item>
               <widget class="QLineEdit" name="input_tcp_client_port">
                <property name="toolTip">
                 <string>Port</string>
                </property>
                <property name="text">
                 <string>50000</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_mode_udp">
             <attribute name="title">
              <string>udp</string>
             </attribute>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLineEdit" name="input_udp_port"/>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_run">
            <property name="toolTip">
             <string>connect</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="autoRepeat">
             <bool>false</bool>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
            <property name="default">
             <bool>false</bool>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="text_recv">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="text_send">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_imu">
       <attribute name="title">
        <string>IMU</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="Settings">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>756</width>
             <height>472</height>
            </rect>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>30</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
